paint/PAINT_ColourPicker.cpp

```diff
@@ -18,32 +18,64 @@ ColourPicker::ColourPicker(gfx::Rectangle rect, SDL_Renderer* renderer, std::sha
+void ColourPicker::initialiseColourValueBoxes()
+{
+	redValueBox_ = std::make_shared<win::ColourValueTextbox>(gfx::Rectangle(), "redValueBox", renderer_, 18, 2, -3, drawWindowPtr_->getDrawRed());
 	redValueBox_->setBackgroundColour({ 255,50,50,200 });
-	greenValueBox_ = std::make_shared<win::ColourValueTextbox>(gfx::Rectangle(), "greenValueBox", renderer, 18, 2, -3, drawWindowPtr_->getDrawGreen());
+	greenValueBox_ = std::make_shared<win::ColourValueTextbox>(gfx::Rectangle(), "greenValueBox", renderer_, 18, 2, -3, drawWindowPtr_->getDrawGreen());
 	greenValueBox_->setBackgroundColour({ 50,255,50,200 });
-	blueValueBox_ = std::make_shared<win::ColourValueTextbox>(gfx::Rectangle(), "blueValueBox", renderer, 18, 2, -3, drawWindowPtr_->getDrawBlue());
+	blueValueBox_ = std::make_shared<win::ColourValueTextbox>(gfx::Rectangle(), "blueValueBox", renderer_, 18, 2, -3, drawWindowPtr_->getDrawBlue());
 	blueValueBox_->setBackgroundColour({ 50, 50, 255, 200 });
-	alphaValueBox_ = std::make_shared<win::ColourValueTextbox>(gfx::Rectangle(), "alphaValueBox", renderer, 18, 2, -3, drawWindowPtr_->getDrawAlpha());
+	alphaValueBox_ = std::make_shared<win::ColourValueTextbox>(gfx::Rectangle(), "alphaValueBox", renderer_, 18, 2, -3, drawWindowPtr_->getDrawAlpha());
 	alphaValueBox_->setBackgroundColour({ 150,150,150,200 });
-	
 	colourValuesBox_->addChild(redValueBox_);
 	colourValuesBox_->addChild(greenValueBox_);
 	colourValuesBox_->addChild(blueValueBox_);
 	colourValuesBox_->addChild(alphaValueBox_);
-
 	addChild(colourValuesBox_);
+}
+
+
+void ColourPicker::initialiseColourSliders()
+{
+	redValueSlider_ = std::make_shared<win::ColourSlider>(renderer_, gfx::Rectangle(15, 660, 105, 20), "redValueSlider", gfx::Colour(255, 255, 255, 255), gfx::Colour(0, 0, 0, 255), drawWindowPtr_->getDrawRed());
+	greenValueSlider_ = std::make_shared<win::ColourSlider>(renderer_, gfx::Rectangle(15, 660, 105, 20), "greenValueSlider", gfx::Colour(255, 255, 255, 255), gfx::Colour(0, 0, 0, 255), drawWindowPtr_->getDrawGreen());
+	blueValueSlider_ = std::make_shared<win::ColourSlider>(renderer_, gfx::Rectangle(15, 660, 105, 20), "blueValueSlider", gfx::Colour(255, 255, 255, 255), gfx::Colour(0, 0, 0, 255), drawWindowPtr_->getDrawBlue());
+	alphaValueSlider_ = std::make_shared<win::ColourSlider>(renderer_, gfx::Rectangle(15, 660, 105, 20), "alphaValueSlider", gfx::Colour(255, 255, 255, 255), gfx::Colour(0, 0, 0, 255), drawWindowPtr_->getDrawAlpha());
+
+	colourSliders_->addChild(redValueSlider_);
+	colourSliders_->addChild(greenValueSlider_);
+	colourSliders_->addChild(blueValueSlider_);
+	colourSliders_->addChild(alphaValueSlider_);
+	addChild(colourSliders_);
+	colourSliders_->applyLayout();
+	redValueSlider_->updateLineMarker();
+	greenValueSlider_->updateLineMarker();
+	blueValueSlider_->updateLineMarker();
+	alphaValueSlider_->updateLineMarker();
 
-	setDrawColourFromActive();
 }
 
 void ColourPicker::setDrawColourFromActive() const
```

Comment:
Should probably prefer cout to printf. Then again, line 84 looks like a logic condition that should never occur => assertion. Remove or comment out or use a dissolving macro for diagnostic code.
**MUST**

Reply:
changed to assert to check both are not active/inactive at same time. removed other cout/printf here.